cmake/modules/RootConfiguration.cmake

```diff
@@ -718,9 +721,11 @@ if(WIN32)
   # We cannot use the compiledata.sh script for windows
   configure_file(${CMAKE_SOURCE_DIR}/cmake/scripts/compiledata.win32.in ${CMAKE_BINARY_DIR}/ginclude/compiledata.h NEWLINE_STYLE UNIX)
 else()
+  # Needed by ACLIC, while in ROOT we are using everywhere C++ standard via CMake features that are requested to build target
+  set(CMAKE_CXX_ACLIC_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION}")
```

Comment:
Event without this, i.e. in current master, my `ginclude/comiledata.h` contains `-std=c++17`. So this doesn't seem needed?

Reply:
Yes, seems like it! I reverted this change.

Reply:
Actually it was necessary. Without this PR, you are right, it contains it because of this hack in `CheckCompiler.cmake`:
```cmake
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION}")
```

But this PR is removing that hack, so now the standard is not in the `CMAKE_CXX_FLAGS` anymore and we need to add it back for `compiledata.sh`.